delivery/client.py

```diff
@@ -200,18 +201,41 @@ def update_metadata(
         self,
         data: typing.Iterable[dso.model.ArtefactMetadata],
     ):
-        res = self.session.put(
-            url=self._routes.update_metadata(),
-            json={'entries': [
-                dataclasses.asdict(
-                    artefact_metadata,
-                    dict_factory=ci.util.dict_to_json_factory,
-                ) for artefact_metadata in data
-            ]},
-            timeout=(4, 121),
+        # update metadata by type to reduce the per-request-size to lower risks of running into
+        # http 413 errors. However, don't split data of the same type into different requests
+        # because the delivery services requires all data of the same type to be in the same
+        # request (the request is being used as a kind of context, so that not containing elements
+        # are being removed from the delivery database)
+        data_by_types = itertools.groupby(
+            sorted(data, key=lambda datum: datum.meta.type),
```

Comment:
why do we need to sort previously to grouping?

Reply:
Because the `itertools.groupby` function only groups consecutive items together, so you have to sort first to establish this order and then you can group by each value.